v0/destinations/fbpixel/transform.js

```diff
@@ -0,0 +1,431 @@
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.USERDATA.name]
+  );
+  let custom_data = {};
+
+  const commonData = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.COMMON.name]
+  );
+  if (category.type !== "identify") {
+    custom_data = {
+      ...custom_data,
+      ...flattenJson(constructPayload(message, MAPPING_CONFIG[category.name]))
+    };
+    custom_data = checkPiiProperties(
+      message,
+      custom_data,
+      blacklistPiiProperties,
+      whitelistPiiProperties,
+      category.standard,
+      eventCustomProperties
+    );
+  } else {
+    custom_data = undefined;
+  }
+  if (category.standard) {
+    custom_data.currency = message.properties.currency || "USD";
+    switch (category.type) {
+      case "product list viewed":
+        custom_data = {
+          ...custom_data,
+          ...handleProductListViewed(message, categoryToContent)
+        };
+        commonData.event_name = "ViewContent";
+        break;
+      case "product viewed":
+        custom_data = {
+          ...custom_data,
+          ...handleProduct(message, categoryToContent, valueFieldIdentifier)
+        };
+        commonData.event_name = "ViewContent";
+        break;
+      case "product added":
+        custom_data = {
+          ...custom_data,
+          ...handleProduct(message, categoryToContent, valueFieldIdentifier)
+        };
+        commonData.event_name = "AddToCart";
+        break;
+      case "order completed":
+        custom_data = {
+          ...custom_data,
+          ...handleOrder(message, categoryToContent, valueFieldIdentifier)
+        };
+        commonData.event_name = "Purchase";
+        break;
+      case "products searched":
+        custom_data = {
+          ...custom_data,
+          search_string: message.properties.query
+        };
+        commonData.event_name = "Search";
+        break;
+      case "checkout started":
+        custom_data = {
+          ...custom_data,
+          ...handleOrder(message, categoryToContent, valueFieldIdentifier)
+        };
+
+        commonData.event_name = "InitiateCheckout";
+        break;
+      default:
+        throw Error("This standard event does not exist");
+    }
+  }
+
+  if (category.type === "page") {
+    commonData.event_name = message.name
+      ? `Viewed Page ${message.name}`
+      : "Viewed a Page";
+  }
+  if (category.type === "simple track") {
+    custom_data.value = message.properties
+      ? message.properties.revenue
+      : undefined;
+    delete custom_data.revenue;
+  }
+
+  if (limitedDataUSage) {
+    const dpo = message.integrations.FacebookPixel
+      ? message.integrations.FacebookPixel.dataProcessingOptions
+      : undefined;
+    [
+      commonData.data_processing_options,
+      commonData.data_processing_options_country,
+      commonData.data_processing_options_state
+    ] = dpo;
+  }
+
+  if (user_data && commonData) {
+    const split = user_data.name ? user_data.name.split(" ") : null;
```

Comment:
let's have this logic at the place where we are construction user_data above

Reply:
yes done.

Reply:
done